### PR TITLE
Message logs

### LINKED ISSUE #30: Message Logs
```
As a: Moderator
I want to: see message logs
So that: I can see if anything suspicious is being said
```

When a message is edited, it should be logged to the `#logs` channel (ID: `405068878151024640`) with the previous content, new content and author:
message-updated
This should be done in`LogMessageUpdateHandler`.

When a message is deleted, logged to the `#logs` channel (ID: `405068878151024640`) with the previous content and author. This should be done in`LogMessageDeleteHandler`.

### PR DESCRIPTION
**Pull request by jacobgasser**
# Message Events
Resolves #30 
## Summary
This PR adds two new handlers `LogMessageUpdateHandler` and `LogMessageDeleteHandler` which act on their respective discord events. The primary goal for this is to log when a user has edited their message or deleted a message into the logs channel, this can aid in things like moderation of the server.

### CODE REVIEW

File: test/event/handlers/LogMessageDeleteHandlerTest.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        let sandbox: SinonSandbox;
+        let handler: EventHandler;
+        let discordMock: MockDiscord;
+
+        beforeEach(() => {
+            sandbox = createSandbox();
+            handler = new LogMessageUpdateHandler();
+            discordMock = new MockDiscord();
+        });
+
```
Comment (LamboCreeper) [inline test/event/handlers/LogMessageDeleteHandlerTest.ts:27]: Can you remove this line break, please? Only one gap needed between the blocks.

File: test/event/handlers/LogMessageDeleteHandlerTest.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        it("sends a message in logs channel when a message is deleted", async () => {
+            const message = discordMock.getMessage();
+            const messageMock = sandbox.stub(message.guild.channels.cache, "find");
+
+            message.content = "message content";
+
+            await handler.handle(message);
+
+            const messageChannel = await messageMock;
```
Comment (LamboCreeper) [inline test/event/handlers/LogMessageDeleteHandlerTest.ts]: What does this do? You're not using the `messageChannel` constant anywhere.

File: test/event/handlers/LogMessageUpdateHandlerTest.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            const oldMessage = discordMockOld.getMessage();
+            const newMessage = discordMockNew.getMessage();
+            const messageMock = sandbox.stub(oldMessage.guild.channels.cache, "find");
+
+            oldMessage.content = "oldMessage content";
+            newMessage.content = "newMessage content";
+
+            await handler.handle(oldMessage, newMessage);
+
+            const messageChannel = await messageMock;
```
Comment (LamboCreeper) [inline test/event/handlers/LogMessageUpdateHandlerTest.ts]: What does this do? You're not using the `messageChannel` constant anywhere.

**Review by LamboCreeper — CHANGES_REQUESTED** [review]
Looks good, congrats on writing your first tests and TypeScript. Just a couple of comments to be addressed.

File: src/event/handlers/LogMessageDeleteHandler.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { Constants, MessageEmbed, Message, TextChannel } from "discord.js";
+import { LOG_CHANNEL_ID } from "../../config.json";
+import EventHandler from "../../abstracts/EventHandler";
+
+class LogMessageUpdateHandler extends EventHandler {
```
Comment (Jason2605) [inline src/event/handlers/LogMessageDeleteHandler.ts]: This should be `LogMessageDeleteHandler`, same for the export.

Comment (jacobgasser) [reply src/event/handlers/LogMessageDeleteHandler.ts]: done

File: test/event/handlers/LogMessageUpdateHandlerTest.ts (hunk 3 above)
Comment (jacobgasser) [reply test/event/handlers/LogMessageUpdateHandlerTest.ts]: removed

File: test/event/handlers/LogMessageDeleteHandlerTest.ts (hunk 2 above)
Comment (jacobgasser) [reply test/event/handlers/LogMessageDeleteHandlerTest.ts]: removed

File: test/event/handlers/LogMessageUpdateHandlerTest.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import { expect } from "chai";
+import { Constants, Message } from "discord.js";
+import LogMessageUpdateHandler from "../../../src/event/handlers/LogMessageUpdateHandler";
+import { SinonSandbox, createSandbox } from "sinon";
+import EventHandler from "../../../src/abstracts/EventHandler";
+import MockDiscord from "../../MockDiscord";
+
+
```
Comment (LamboCreeper) [inline test/event/handlers/LogMessageUpdateHandlerTest.ts]: Can you remove this line break, please? Only one gap needed between the blocks.

**Review by LamboCreeper — CHANGES_REQUESTED** [review]

**Review by LamboCreeper — APPROVED** [review]

**Review by Jason2605 — APPROVED** [review]

**Pull request merged**